nc-news-react/src/components/SingleArticle.jsx

```diff
@@ -1,30 +1,41 @@
 import { useEffect, useState } from "react"
 import { Link, useParams } from "react-router-dom"
-import { getArticleById } from "../axios"
+import { getArticleById, increaseArticleVotes } from "../axios"
 
 function SingleArticle () {
 
     const {article_id} = useParams()
     const [article, setArticle] = useState({})
     const [loading, setLoading] = useState(true)
+    const [votes, setVotes] = useState('')
 
     useEffect(()=> {
         getArticleById(article_id).then((articleData)=> {
             setArticle(articleData)
             setLoading(false)
         })
-    },[article_id])
+    },[article_id, votes])
 
     if(loading === true){return <p>Loading...</p>}
 
 
+    function handleVotes() {
+        increaseArticleVotes(article_id).then((response)=> {
+            setVotes(response.votes)
+    })}
+    
```

Comment:
This does work, but we would like you to use optimistic rendering for this. Currently, every time someone clicks vote, it will rerender the article component. Imaging hundreds of users doing this! Optimistic rendering solves this and also looks slicker!